src/Compilers/CSharp/Test/Symbol/Symbols/SymbolEqualityTests.cs

```diff
@@ -911,6 +922,164 @@ public static void M(A<T> t)
                 );
         }
 
+        [Fact]
+        public void TestImplementationAttributes_Inlining()
+        {
+            var src = @"
+using System.Runtime.CompilerServices;
+
+public class C
+{
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public void M_Aggressive()
+    {
+    }
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    public void M_NoInlining()
+    {
+    }
+}
+";
+            var comp = CreateCompilation(src);
+            comp.VerifyDiagnostics();
+
+            var aggressiveInliningMethod = (IMethodSymbol)comp.GetSymbolsWithName("M_Aggressive", SymbolFilter.Member).Single();
```

Comment:
We need to verify these from both source and metadata. Instead of getting the symbol from the compilation like this, create a helper that, given a ModuleSymbol, finds and verifies the symbols. You can then pass that as the `symbolValidator` and `sourceSymbolValidator` parameters to `CompileAndVerify`. Same for the test of these added tests.

Reply:
@333fred I updated one test. Can you check whether the change is correct? If correct, I'll update the rest of them. Otheriwse, I'll need some guidance. Thanks.

Reply:
@Youssef1313 looks correct. The only comment is that `CompileAndVerify` also ensures there are no diagnostics, so you can remove the `CreateCompilation` and `VerifyDiagnostics` pair.